DQMOffline/Trigger/plugins/Tau3MuMonitor.cc

```diff
@@ -0,0 +1,153 @@
+#include "DQMOffline/Trigger/plugins/Tau3MuMonitor.h"
+
+Tau3MuMonitor::Tau3MuMonitor( const edm::ParameterSet& iConfig ) : 
+  folderName_         ( iConfig.getParameter<std::string>("FolderName") ) ,
+  tauToken_           ( mayConsume<reco::CompositeCandidateCollection>(iConfig.getParameter<edm::InputTag>("taus") ) ) ,
+  pt_binning_         ( getHistoPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>("ptPSet"  ) ) ) ,
+  eta_binning_        ( getHistoPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>("etaPSet" ) ) ) ,
+  phi_binning_        ( getHistoPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>("phiPSet" ) ) ) ,
+  mass_binning_       ( getHistoPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>("massPSet") ) ) ,
+  genTriggerEventFlag_( new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("GenericTriggerEventPSet"),consumesCollector(), *this) )
+{
+  // initialise histograms to null
+  tau1DPt_     = nullptr;
+  tau1DEta_    = nullptr;
+  tau1DPhi_    = nullptr;
+  tau1DMass_   = nullptr;
+  tau2DEtaPhi_ = nullptr;
+}
+
+Tau3MuMonitor::~Tau3MuMonitor()
+{
+  if (genTriggerEventFlag_) delete genTriggerEventFlag_;
+}
+
+// shape the content of a "histogram PSet"
+void Tau3MuMonitor::fillHistoPSetDescription(edm::ParameterSetDescription & pset)
+{
+  pset.add<unsigned int>( "nbins" );
+  pset.add<double>      ( "xmin"  );
+  pset.add<double>      ( "xmax"  );
+}
+
+// read the information packed into an "histogram PSet"
+MEbinning Tau3MuMonitor::getHistoPSet(edm::ParameterSet pset)
+{
+  return MEbinning{
+    pset.getParameter<unsigned int>("nbins"),
+    pset.getParameter<double >     ("xmin" ),
+    pset.getParameter<double >     ("xmax" ),
+  };
+}
+
+// book histograms
+void Tau3MuMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+                                   edm::Run const        & iRun,
+                                   edm::EventSetup const & iSetup) 
+{  
+  std::string histname;
+
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+  
+  // tau 3 mu 1D pt
+  histname = "tau1DPt"; 
+  tau1DPt_ = ibooker.book1D(histname, "", pt_binning_.nbins, pt_binning_.xmin, pt_binning_.xmax);
```

Comment:
to make it readable it would require to divide the entries in each bin by the bin width.
It's probably possible but it's added complication